user-page/user_homepage/script.js

```diff
@@ -0,0 +1,175 @@
+//fake data used throughout the script for testing
```

Comment:
You did a lot of work on your own, formatting the time. I suggest to read [this](https://www.toptal.com/software/definitive-guide-to-datetime-manipulation) when working with time, maybe an existing library would satisfy your needs.

Reply:
I actually thought about standardizing time in JS format. This was after I wrote all the functions. I may use this solution in the future.